recipes/fast-cdr/all/conanfile.py

```diff
@@ -0,0 +1,101 @@
+    url = "https://github.com/conan-io/conan-center-index"
+    description = "eProsima FastCDR library for serialization"
+    topics = ("conan","DDS", "Middleware","Serialization")
+    settings = "os", "compiler", "build_type", "arch"
+    options = {
+        "shared":          [True, False],
+        "fPIC":            [True, False]
+    }
+    default_options = {
+        "shared":            False,
+        "fPIC":              True
+    }
+    exports_sources = ["patches/**"]
+    _cmake = None
+
+    @property
+    def _pkg_cmake(self):
+        return os.path.join(
+            self.package_folder,
+            "lib",
+            "cmake"
+        )
+
+    @property
+    def _pkg_share(self):
+        return os.path.join(
+            self.package_folder,
+            "share"
+        )
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    def _patch_sources(self):
+        for patch in self.conan_data["patches"][self.version]:
+            tools.patch(**patch)
+
+    def _get_configured_cmake(self):
+        if self._cmake:
+            pass 
+        else:
+            self._cmake = CMake(self)
+        self._cmake.configure(
+            source_dir = self._source_subfolder
+        )
+        return self._cmake
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version], strip_root=True,
+                  destination=self._source_subfolder)
+    
+    def configure(self):
+        if self.options.shared:
+            del self.options.fPIC
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def validate(self):
+        if self.settings.os == "Linux":
+            if (self.settings.compiler == "gcc" and self.settings.compiler.libcxx == "libstdc++"):
+                raise ConanInvalidConfiguration("This package requires libstdc++11 for libcxx setting")
+            if (self.settings.compiler == "clang" and self.settings.compiler.libcxx == "libstdc++"):
+                raise ConanInvalidConfiguration("This package requires libstdc++11 for libcxx setting")
+            if (self.settings.compiler == "clang" and self.settings.compiler.libcxx == "libc++"):
+                raise ConanInvalidConfiguration("This package requires libstdc++11 for libcxx setting")
+        if self.settings.os == "Windows":
+            if self.settings.compiler == "Visual Studio":
+                raise ConanInvalidConfiguration("This package dont not compile with Visual Studio.")
+
+    def build(self):
+        self._patch_sources()
+
+        cmake = self._get_configured_cmake()
+        cmake.build()
+
+    def package(self):
+        cmake = self._get_configured_cmake()
+        cmake.install()
+        self.copy("LICENSE", src=self._source_subfolder, dst="licenses")
+        tools.rmdir(self._pkg_cmake)
+        tools.rmdir(self._pkg_share)
+        
+    def package_info(self):
```

Comment:
We have a method for doing this https://github.com/conan-io/conan-center-index/pull/5581/files#diff-039ac6cecc39d15aef646db0a407ac4a2e6d1ee07120595db6deab9e07665d0fR71-R81

Which was done in the PR 

Otherwise please leave a remark

```suggestion
    def package_info(self):
    # FIXME: FastCDR does not install under a CMake namespace https://github.com/eProsima/Fast-CDR/blob/cff6ea98f66d5fd4d53541e183676257a42a6c23/src/cpp/CMakeLists.txt#L156
```

Reply:
I made like this : its now best practise "fastcdr::fastcdr" and also without namespace "fastcdr" as an alias for other consuming projects (like Fast-DDS which i want to add). by the way - nice hint - i also used the hint for adding properties to a target.